rest_framework/decorators.py

```diff
@@ -120,6 +123,32 @@ def decorator(func):
     return decorator
 
 
+def paginate(view_method):
+    """
+    Adds Pagination to the method
+    """
+    @wraps(view_method)
+    def inner(view, *args, **kwargs):
+        assert isinstance(view, APIView), (
+            "paginate must be applied on an APIView, but applied on %s." % APIView
+        )
+
+        queryset = view_method(view, *args, **kwargs)
+
+        assert isinstance(queryset, (list, QuerySet)), (
+            "paginate expects a list or a QuerySet but got %s." % queryset
```

Comment:
I might use `type(queryset)` instead coercing the actual value to a string. e.g., a model instance might have a `__str__` implementation that doesn't make it obvious that the object is a model.

Reply:
Agreed.